src/rest_api/class-openagenda-controller.php

```diff
@@ -1736,4 +1750,44 @@ function ( $a, $b ) {
 
 		return $posts;
 	}
+
+	/**
+	 * Get latitude and longitude from address.
+	 *
+	 * @param string $address The address.
+	 * @param string $zipcode The zipcode.
+	 * @param string $city The city.
+	 *
+	 * @return array|bool The latitude and longitude.
+	 */
+	public function get_latlng_from_address( $address, $zipcode, $city ) {
+		if ( empty( $address ) || empty( $zipcode ) || empty( $city ) ) {
+			return false;
+		}
+
+		$address_full = $address . ', ' . $zipcode . ' ' . $city;
+		$address_full = rawurlencode( $address_full );
+
+		// Get the address data from OSM (OpenStreetMap).
+		$osm_url     = 'https://nominatim.openstreetmap.org/search?q=' . $address_full . '&format=json&addressdetails=1';
+		$osm_address = wp_remote_get( $osm_url );
+
+		if ( ! $osm_address ) {
+			return null;
+		}
+
+		$osm_address = json_decode( $osm_address['body'] );
+
+		if ( ! $osm_address[0]->lat || ! $osm_address[0]->lon ) {
+			return null;
```

Comment:
osm_address could be something other than an array, with one element, that is an object, with a lat
please consider extra checks

Reply:
I added an extra check